### PR TITLE
Create Kyma installation package

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Changes proposed in this pull request:

- Create installation package fo Kyma

**Related issue(s)**
https://github.com/kyma-incubator/hydroform/issues/33

### CODE REVIEW

File: install/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Install
+
+## Overview
+
+Package `install` contains library used to install Kyma on already existing clusters.
```
Comment (majakurcius) [inline install/README.md]:
```suggestion
The `install` package contains the library used to install Kyma on already existing clusters.
```

File: install/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+## Usage
+
+The installation process consist of two phases:
+- `PrepareInstallation` - in this phase Tiller is installed and Kyma Installer is deployed to the cluster along with the default configuration.
```
Comment (majakurcius) [inline install/README.md]:
```suggestion
- `PrepareInstallation` - in this phase Tiller is installed and the Kyma Installer is deployed to the cluster along with the default configuration.
```

File: install/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- `StartInstallation` - in this phase the Installation is triggered by labeling Installation Custom Resource.
+
+### Example
+
+The example of using the library to install Kyma on properly configured Minikube cluster can be found [here](https://github.com/kyma-incubator/hydroform/tree/master/install/examples/example.go).
```
Comment (majakurcius) [inline install/README.md]:
```suggestion
An example of using the library to install Kyma on a properly configured Minikube cluster can be found [here](https://github.com/kyma-incubator/hydroform/tree/master/install/examples/example.go). 
```

Alternatively:
```
To see how to use the library to install Kyma on a properly configured Minikube cluster, see this [example](https://github.com/kyma-incubator/hydroform/tree/master/install/examples/example.go).

File: install/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## Usage
+
+The installation process consist of two phases:
+- `PrepareInstallation` - in this phase Tiller is installed and the Kyma Installer is deployed to the cluster along with the default configuration.
+- `StartInstallation` - in this phase the Installation is triggered by labeling Installation Custom Resource.
+
+### Example
+
+To see how to use the library to install Kyma on a properly configured Minikube cluster, see this [example](https://github.com/kyma-incubator/hydroform/tree/master/install/examples/example.go).
```
Comment (majakurcius) [inline install/README.md]:
```suggestion
To learn how to use the library to install Kyma on a properly configured Minikube cluster, see this [example](https://github.com/kyma-incubator/hydroform/tree/master/install/examples/example.go).
```

File: Makefile [hunk 5]
```diff
@@ -1,6 +1,8 @@
 .PHONY: build
-build: 
+build:
 	./before-commit.sh ci
+	# TODO: we should provide separate Prow jobs for provisioning and installation packages after the repository will be adjusted
```
Comment (clebs) [inline Makefile]: I am not sure if this is the right approach. This repository is a library and thus generates no binaries, so when releaseing (tagging) one module, we inevitably also release all others as well.
Based on that, I think we should have only one CI pipeline for the complete repository.

Comment (Szymongib) [reply Makefile]: That is a fair point. I only wonder if we will provide some more complex and long-running tests those for provisioning would not need to be run if changes were made to the installation, but I guess we can do a separate job for that letter.
I will remove the TODO then.

Comment (clebs) [reply Makefile]: True, in that case the CI job could be smart and run only the tests of the modules with changes on them.

File: install/installation/configuration.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+	for _, entry := range c {
+		if entry.Key == key {
+			return entry, true
+		}
+	}
+
+	return ConfigEntry{}, false
+}
+
+func (c *ConfigEntries) Set(key, value string, secret bool) {
```
[the remark this answers is not in the document]
Comment (Szymongib) [reply install/installation/configuration.go]: Good catch! Fixed.

File: install/installation/installation.go [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+	}
+
+	restMapper := restmapper.NewDiscoveryRESTMapper(groupResources)
+
+	dynamicClient, err := dynamic.NewForConfig(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	installtionClient, err := installationClientset.NewForConfig(kubeconfig)
```
[the remark this answers is not in the document]
Comment (Szymongib) [reply install/installation/installation.go]: Fixed

File: install/util/wait_test.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+			return false, nil
+		})
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "timeout")
+	})
+
+	t.Run("should return error when error occured", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Szymongib) [reply install/util/wait_test.go]: Fixed

File: install/k8s/generic_client_test.go [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+		client := NewGenericClient(nil, nil, k8sClientSet)
+
+		// when
+		err := client.WaitForPodByLabel(namespace, labelSelector, v1.PodRunning, waitForPodTimeout, waitForPodCheckInterval)
+
+		// then
+		require.Error(t, err)
+	})
+
+	t.Run("should return error if pod does not have correct label", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Szymongib) [reply install/k8s/generic_client_test.go]: Fixed

**Pull request merged**